### PR TITLE
[GEOS-11373] : Geometry type mismatch in WFS 1.1.0

### PR DESCRIPTION
**Pull request by eropartz**
[GEOS-11373](https://osgeo-org.atlassian.net/browse/GEOS-11373) [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=geotools&utm_campaign=badge_info)

  

# Checklist

- [x] I have read the [contribution guidelines](https://github.com/geotools/geotools/blob/main/CONTRIBUTING.md).
- [x] I have sent a [Contribution Licence Agreement](https://docs.geotools.org/latest/developer/procedures/contribution_license.html) (not required for small changes, e.g., fixing typos in documentation).
- [x] First PR targets the `main` branch (backports managed later; ignore for branch specific issues).
- [x] Avoid [Java 9+ split packages](http://tutorials.jenkov.com/java/modules.html#split-packages-not-allowed).
- [x] All the build checks are green ([see automated QA checks](https://docs.geotools.org/latest/developer/conventions/code/qa.html)).

For core and extension modules:

- [x] New unit tests have been added covering the changes.
- [ ] [Documentation](https://github.com/geotools/geotools/tree/main/docs) has been updated (if change is visible to end users).
- [x] There is an issue in [GeoTools Jira](https://osgeo-org.atlassian.net/projects/GEOT) (except for changes not visible to end users). 
- [x] Commit message(s) must be in the form ``[GEOT-XYZW] Title of the Jira ticket``.
- [x] Bug fixes and small new features are presented as a single commit.
- [x] The commit targets a single objective (if multiple focuses cannot be avoided, each one is in its own commit, and has a separate ticket describing it).

### CODE REVIEW

**Comment by jodygarnett** [comment]
This seems an important fix, it requires a test case before I can review it?

**Comment by eropartz** [comment]
Hi @jodygarnett , Unit test added

File: modules/extension/xsd/xsd-gml3/src/main/java/org/geotools/gml3/GMLSchema.java [hunk 1]
```diff
@@ -12969,7 +12970,7 @@ private static AttributeType build_MULTISURFACEPROPERTYTYPE_TYPE() {
         AttributeType builtType =
                 new AttributeTypeImpl(
                         new NameImpl("http://www.opengis.net/gml", "MultiSurfacePropertyType"),
-                        MultiSurface.class,
+                        org.geotools.geometry.jts.MultiSurface.class,
```
Comment (mprins) [inline modules/extension/xsd/xsd-gml3/src/main/java/org/geotools/gml3/GMLSchema.java:12973]: Why use a fully qualified class when it is imported?

**Comment by aaime** [comment]
The backport to `31.x` failed:
```
The process '/usr/bin/git' failed with exit code 1
```
<details>
<summary>stderr</summary>

```
The previous cherry-pick is now empty, possibly due to conflict resolution.
If you wish to commit it anyway, use:

    git commit --allow-empty

Otherwise, please use 'git cherry-pick --skip'

```
</details>

<details>
<summary>stdout</summary>

```
[backport-4773-to-31.x 2617d9425a] Geometry type mismatch in WFS 1.1.0
 Author: ROPARTZ Erwan DTSI/DSI <erwan.ropartz@orange.com>
 Date: Tue May 21 15:06:05 2024 +0200
 1 file changed, 2 insertions(+), 2 deletions(-)
[backport-4773-to-31.x ed694c22c6] Replace org.geotools.gml3.MultiSurface by org.geotools.geometry.jts.MultiSurface
 Author: ROPARTZ Erwan DTSI/DSI <erwan.ropartz@orange.com>
 Date: Thu May 23 09:59:26 2024 +0200
 5 files changed, 30 insertions(+), 37 deletions(-)
 delete mode 100644 modules/extension/xsd/xsd-gml3/src/main/java/org/geotools/gml3/MultiSurface.java
[backport-4773-to-31.x 8eb8b9b468] remove fully qualified class name MultiSurface
 Author: ROPARTZ Erwan DTSI/DSI <erwan.ropartz@orange.com>
 Date: Thu May 23 16:59:45 2024 +0200
 1 file changed, 1 insertion(+), 1 deletion(-)
On branch backport-4773-to-31.x
You are currently cherry-picking commit c8298b15b2.
  (all conflicts fixed: run "git cherry-pick --continue")
  (use "git cherry-pick --skip" to skip this patch)
  (use "git cherry-pick --abort" to cancel the cherry-pick operation)

nothing to commit, working tree clean

```
</details>

To backport manually, run these commands in your terminal:
```bash
# Fetch latest updates from GitHub
git fetch
# Create a new working tree
git worktree add .worktrees/backport-31.x 31.x
# Navigate to the new working tree
cd .worktrees/backport-31.x
# Create a new branch
git switch --create backport-4773-to-31.x
# Cherry-pick the merged commit of this pull request and resolve the conflicts
git cherry-pick 5fdfba3124e009be72f0beb260c7c53cff7d2192,149af2ba19d1b05f69410a759f79bac06610b9cc,8d369eee8a561e88b16f827936292f9c4e34660a,c8298b15b236029cc12186482223b168fbd44ac3
# Push it to GitHub
git push --set-upstream origin backport-4773-to-31.x
# Go back to the original working tree
cd ../..
# Delete the working tree
git worktree remove .worktrees/backport-31.x
```
Then, create a pull request where the `base` branch is `31.x` and the `compare`/`head` branch is `backport-4773-to-31.x`.

**Comment by aaime** [comment]
The backport to `30.x` failed:
```
The process '/usr/bin/git' failed with exit code 1
```
<details>
<summary>stderr</summary>

```
The previous cherry-pick is now empty, possibly due to conflict resolution.
If you wish to commit it anyway, use:

    git commit --allow-empty

Otherwise, please use 'git cherry-pick --skip'

```
</details>

<details>
<summary>stdout</summary>

```
[backport-4773-to-30.x ca4d7fbffd] Geometry type mismatch in WFS 1.1.0
 Author: ROPARTZ Erwan DTSI/DSI <erwan.ropartz@orange.com>
 Date: Tue May 21 15:06:05 2024 +0200
 1 file changed, 2 insertions(+), 2 deletions(-)
[backport-4773-to-30.x ee463f25be] Replace org.geotools.gml3.MultiSurface by org.geotools.geometry.jts.MultiSurface
 Author: ROPARTZ Erwan DTSI/DSI <erwan.ropartz@orange.com>
 Date: Thu May 23 09:59:26 2024 +0200
 5 files changed, 30 insertions(+), 37 deletions(-)
 delete mode 100644 modules/extension/xsd/xsd-gml3/src/main/java/org/geotools/gml3/MultiSurface.java
[backport-4773-to-30.x 640535777a] remove fully qualified class name MultiSurface
 Author: ROPARTZ Erwan DTSI/DSI <erwan.ropartz@orange.com>
 Date: Thu May 23 16:59:45 2024 +0200
 1 file changed, 1 insertion(+), 1 deletion(-)
On branch backport-4773-to-30.x
You are currently cherry-picking commit c8298b15b2.
  (all conflicts fixed: run "git cherry-pick --continue")
  (use "git cherry-pick --skip" to skip this patch)
  (use "git cherry-pick --abort" to cancel the cherry-pick operation)

nothing to commit, working tree clean

```
</details>

To backport manually, run these commands in your terminal:
```bash
# Fetch latest updates from GitHub
git fetch
# Create a new working tree
git worktree add .worktrees/backport-30.x 30.x
# Navigate to the new working tree
cd .worktrees/backport-30.x
# Create a new branch
git switch --create backport-4773-to-30.x
# Cherry-pick the merged commit of this pull request and resolve the conflicts
git cherry-pick 5fdfba3124e009be72f0beb260c7c53cff7d2192,149af2ba19d1b05f69410a759f79bac06610b9cc,8d369eee8a561e88b16f827936292f9c4e34660a,c8298b15b236029cc12186482223b168fbd44ac3
# Push it to GitHub
git push --set-upstream origin backport-4773-to-30.x
# Go back to the original working tree
cd ../..
# Delete the working tree
git worktree remove .worktrees/backport-30.x
```
Then, create a pull request where the `base` branch is `30.x` and the `compare`/`head` branch is `backport-4773-to-30.x`.

**Comment by aaime** [comment]
Automatic backports failed. Also a note, I had to make a change in the GeoSever "features-templating" community module as it depended on the class that has just been removed.

**Pull request merged**